Deprecate outdated API link

NobodysNightmare opened the pull request:
Having a separate "ensure connection" link does not add any value, because even the original "open" link does redirect unconnected users properly. On the other hand, immediately returning an ensure_connection URL is only possible if the storage was configured for mutual OAuth 2 authentication, not for authentication through an SSO provider.

# Ticket
https://community.openproject.org/wp/63691

NobodysNightmare committed:
Deprecate outdated API link

Having a separate "ensure connection" link does not add
any value, because even the original "open" link does redirect
unconnected users properly. On the other hand, immediately returning
an ensure_connection URL is only possible if the storage was configured
for mutual OAuth 2 authentication, not for authentication through an
SSO provider.

ba1ash reviewed (approved):
LGTM
Only tests are red at the moment.

NobodysNightmare commented:
> Only tests are red at the moment.

I understood one of the conditions tested by the spec too late. I thought an early return from the representer code was being tested (but it's not). Instead an early return in the method I removed was being tested. Thus I removed the failing test as well. (because it tested the method that was removed)
